plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -128,10 +140,39 @@ export class HogTransformerService {
                 const transformationsFailed: string[] = event.properties?.$transformations_failed || []
                 const transformationsSkipped: string[] = event.properties?.$transformations_skipped || []
 
-                // For now, execute each transformation function in sequence
+                // Get states for all functions to check if any are disabled - only if feature flag is enabled
+                let states: Record<string, { state: HogWatcherState }> = {}
+                if (this.hub.TRANSFORM_EVENT_HOG_WATCHER_ENABLED) {
+                    const timer = transformEventHogWatcherLatency.startTimer()
+                    states = await this.hogWatcher.getStates(teamHogFunctions.map((hf) => hf.id))
+                    transformEventHogWatcherLatency.observe({ operation: 'getStates' }, timer() * 1000) // Convert seconds to milliseconds
```

Comment:
question: is this how the timer API works? I thought you're supposed to pass the labels to the `startTimer` call and just call the returned callback.